jekyll/_cci2/vs-code-extension-overview.adoc

```diff
@@ -188,30 +188,99 @@ image::{{site.baseurl}}/assets/img/docs/vscode-ext-config-test-run-crop.png[Scre
 
-[#steps]
-=== Steps
 
 [WARNING]
 ====
-Triggering a pipeline with an _unversioned_ configuration can have side effects, for example, unwanted deployment. We recommend using this feature **only** on feature branches.
+Use caution when running pipelines with _unversioned_ configuration. Take care to ensure no jobs with unintended effects are included in the run. For more details, see <<security-implications>>.
+
+Access to this feature is controlled by an opt-in setting which only organization admins can control. Once your organization has opted-in, you can opt-out for each individual project. For more details, see <<feature-controls>>. 
 
-Test run pipelines triggered from VS Code have access to contexts, and project environment variables, based on which branch they are run on, and user permissions.
 ====
 
+[#steps]
+=== Steps
+
 The following steps show how to test run a pipeline from the VS Code extension:
 
 . Open VS Code and select the CircleCI logo in the sidebar. The **Pipelines** and **Run** panels will be visible.
 . Expand the **Run** panel to view your projects.
 . For each project, there is a button to trigger the config test run, and check boxes for each workflow and job for you to choose which parts of your config you want to run. Select your options, and click icon:play[]. Any local change you make to your `.circleci/config.yml` file will be included in the test run.
-. The pipeline is now visible in your link:https://app.circleci.com/pipelines/[dashboard] in the CircleCI web app, and the Pipelines panel in the VS Code extension.
+. The pipeline is now visible in your link:https://app.circleci.com/pipelines/[dashboard] in the CircleCI web app, and the Pipelines panel in the VS Code extension. Please note that, for now, pipelines run with unversioned configuration are not differentiated from other pipelines in the CircleCI web app.
+
+[#feature-controls]
+=== Feature Controls
+
+The ability to trigger pipelines from VS Code can be controlled **at the org level** and **at the project level**. 
+
+* Organization level settings can be found under menu:Organization Settings[Advanced]. The setting default is **Off**. Toggle the "Trigger pipelines with unversioned config" option to On to opt-in. 
+    Organization level settings can be changed only by organization admins, and override project settings.
+
+* Project level settings can be found under menu:Project Settings[VS Code integration].  The setting default is **On**. Toggle the "Trigger pipelines with unversioned config" option to Off to opt-out.
```

Comment:
@BeFunes So is this a setting specific to the ability to use the feature through the VS Code extension, or will this also affect the ability to trigger with unversioned config via the API? If it's for both it seems like the menu position should be more general, if that makes sense?

Reply:
It's for both. 
Yes, it makes sense. We decided to put it in "VS Code Integrations" because at the moment it's the only place where we make this feature evident, and we thought this is the place where users would most likely look.
The "Advanced" page of proejct settings is so busy that we worried this wouldn't have been found. 

That said, I agree this should not be the permanent location of this setting, if Open Preview is validated and we decide to move forward and document it through the API etc